components/file/src/main/java/org/trellisldp/file/FileResource.java

```diff
@@ -78,6 +79,11 @@ public Instant getModified() {
         return asLiteral(DC.modified).map(Instant::parse).orElse(null);
     }
 
+    @Override
+    public String getRevision() {
+        return md5Hex(getModified().getNano() + "." + identifier);
```

Comment:
Why hash this? It's not getting any shorter. If two `getModified().getNano() + "." + identifier`s collide, so will their hash, so we're not getting anything other than opacity…

Reply:
Hashing is both for opacity and for conformance. The [spec](https://tools.ietf.org/html/rfc7232#section-2.3) explicitly calls out opacity and limits the valid characters to be ASCII. This way, if someone creates a resource with a unicode character (perfectly valid), it won't wreck the ETag handler.

Reply:
Whoa, wait! I did not understand this new method to be required to produce legal ETags and nothing in the Javadocs in this PR says so. Is that the requirement you are creating here?

Reply:
That would be part of what's going on. But you raise an important point -- we could easily move the hashing part into the HTTP layer and set a `default` method for `Resource::getRevision` simply as a concatenation of `Resource::getModified` and `Resource::getIdentifier`

Reply:
Absolutely (and on my current understanding, preferable). Here are some options:
1. Keep an injectable ETag generator that is smarter.
    * Pro: keeps all the HTTP-standard-specific  stuff in one place, `trellis-http`, con: that generator has to get smarter and smarter for every new use case.
2. What you have here (while perhaps renaming `getRevision` to `getETag` or at least documenting it more explicitly)
    * Pro: keeps the backend-specific smarts in the backend where they belong, con: forces everyone to repeat at least some logic for ETags.
3. What you just suggested (let `Resource`s return the raw material for ETags and let the HTTP layer combine/transform it to ensure legal syntax)
    * Pro: separates responsibilities nicely, keeps things DRY, con: not much that I can see.